src/main/java/fr/openent/viescolaire/service/impl/DefaultServicesService.java

```diff
@@ -216,7 +227,33 @@ public void deleteService(JsonObject oService, Handler<Either<String, JsonObject
         JsonArray values = new JsonArray().add(oService.getString("id_matiere"))
                 .add(oService.getString("id_enseignant")).addAll(classOrGroupIds);
 
-        Sql.getInstance().prepared(query, values, validUniqueResultHandler(handler));
+        if (isNull(moduleServices) || moduleServices.getBoolean(Field.COMPETENCES)) {
+            sql.prepared(query, values, event -> {
+                if (event.body().getString(Field.STATUS).equals(Field.OK)) {
+                    deleteSubtopicsOfService(oService, handler);
+                }
+            });
+        } else {
+            sql.prepared(query, values, validUniqueResultHandler(handler));
+        }
+    }
+
+    private void deleteSubtopicsOfService(JsonObject oService, Handler<Either<String, JsonObject>> requestHandler) {
+        JsonObject action = new JsonObject()
+                .put("action", "services.deleteSubtopics")
+                .put("id_matiere", oService.getString("id_matiere"))
+                .put("id_enseignant", oService.getString("id_enseignant"))
+                .put("id_groups", oService.getJsonArray("id_groups"));
+
+        this.eb.request(Viescolaire.COMPETENCES_BUS_ADDRESS, action, handlerToAsyncHandler(event -> {
+            if (event.body().getString(Field.STATUS).equals(Field.OK)) {
+                requestHandler.handle(new Either.Right<>(new JsonObject().put(Field.RESULTS, event.body().getJsonArray(Field.RESULTS))));
+            } else {
+                log.error(String.format("[Viescolaire@%s::deleteSubtopicsOfService] Failed to send subtopics infos to delete",
+                        this.getClass().getSimpleName()));
+                requestHandler.handle(new Either.Left<>(event.body().getString("message")));
+            }
+        }));
```

Comment:
Voici ce que je te propose : 

```suggestion
  private Future<JsonObject> deleteSubtopicsOfService(JsonObject oService) {
        Promise<JsonObject> promise = Promise.promise();

        JsonObject action = new JsonObject()
                .put("action", "services.deleteSubtopics")
                .put("id_matiere", oService.getString("id_matiere"))
                .put("id_enseignant", oService.getString("id_enseignant"))
                .put("id_groups", oService.getJsonArray("id_groups"));

        this.eb.request(Viescolaire.COMPETENCES_BUS_ADDRESS, action, handlerToAsyncHandler(event -> {
            if (event.body().getString(Field.STATUS).equals(Field.OK)) {
                promise.complete(new JsonObject().put(Field.RESULTS, event.body().getJsonArray(Field.RESULTS)));
            } else {
                log.error(String.format("[Viescolaire@%s::deleteSubtopicsOfService] Failed to send subtopics infos to delete",
                        this.getClass().getSimpleName()));
                promise.fail(event.body().getString(Field.MESSAGE));
            }
        }));
```